### PR TITLE
Organize add-on (& maybe roles) to files

### PR DESCRIPTION
**Pull request by Ultradragon005**
I lost count/like 52

### CODE REVIEW

File: Modules/GameOptionsSender/PlayerGameOptionsSender.cs [hunk 1]
```diff
@@ -507,7 +507,13 @@ public override IGameOptions BuildGameOptions()
...
         {
-            Main.KillGhoul.Add(player.PlayerId);
+            var pc = Utils.GetPlayerById(player.PlayerId);
+            if (pc != null) 
+            {
+                if (!pc.Is(CustomRoles.Pestilence))
+                    Ghoul.KillGhoul.Add(player.PlayerId);
+            }
+            
         }
```
Comment (Tommy-XL) [inline Modules/GameOptionsSender/PlayerGameOptionsSender.cs:518]: This code can be done like:

```
if (Ghoul.IsEnable)
    Ghoul.ApplyGameOptions();
```

**Comment by Tommy-XL** [comment]
image

Same here, but like:

case CustomRoles.Ghoul when (CompletedTasksCount + 1) >= AllTasksCount:
     Ghoul.OnTaskComplete(player);
     break;

File: Modules/GameOptionsSender/PlayerGameOptionsSender.cs [hunk 2]
```diff
@@ -547,13 +554,13 @@ public override IGameOptions BuildGameOptions()
...
+                    opt.SetFloat(FloatOptionNames.ImpostorLightMod, Torch.TorchVision.GetFloat());
 
-                    if (Utils.IsActive(SystemTypes.Electrical) && !Options.TorchAffectedByLights.GetBool())
+                    if (Utils.IsActive(SystemTypes.Electrical) && !Torch.TorchAffectedByLights.GetBool())
                         opt.SetVision(true);
-                    opt.SetFloat(FloatOptionNames.CrewLightMod, Options.TorchVision.GetFloat() * 5);
-                    opt.SetFloat(FloatOptionNames.ImpostorLightMod, Options.TorchVision.GetFloat() * 5);
+                    opt.SetFloat(FloatOptionNames.CrewLightMod, Torch.TorchVision.GetFloat() * 5);
+                    opt.SetFloat(FloatOptionNames.ImpostorLightMod, Torch.TorchVision.GetFloat() * 5);
                     break;
```
Comment (Tommy-XL) [inline Modules/GameOptionsSender/PlayerGameOptionsSender.cs:564]: Same here

**Comment by Ultradragon005** [comment]
mhm I see, I was too focused changing the variables I forgot what individual classes are even made for 😂

File: Roles/AddOns/Crewmate/ghoul.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    public static void SetupCustomOptions()
+    {
+        SetupAdtRoleOptions(Id, CustomRoles.Ghoul, canSetNum: true, tab: TabGroup.Addons);
+    }
+
+    public static void Init()
+    {
+        KillGhoul = [];
+        IsEnable = true;
+    }
```
Comment (Tommy-XL) [inline Roles/AddOns/Crewmate/ghoul.cs:24]: There should be `IsEnable = false;`

File: Patches/PlayerControlPatch.cs [hunk 4]
```diff
@@ -814,7 +796,7 @@ public static bool Prefix(PlayerControl __instance, [HarmonyArgument(0)] PlayerC
 
                 case CustomRoles.Unlucky:
                     var Ue = IRandom.Instance;
-                    if (Ue.Next(1, 100) <= Options.UnluckyKillSuicideChance.GetInt())
+                    if (Ue.Next(1, 100) <= Unlucky.UnluckyKillSuicideChance.GetInt())
                     {
                         Main.PlayerStates[killer.PlayerId].deathReason = PlayerState.DeathReason.Suicide;
                         killer.RpcMurderPlayerV3(killer);
```
Comment (Ultradragon005) [inline Patches/PlayerControlPatch.cs:802]: not sure what to use for whetever unlucky is dead in prefix.. so for now I don't use the method I made here nor shipstatus ventpatch.

Comment (Tommy-XL) [reply Patches/PlayerControlPatch.cs:802]: You can make
`if (!Unlucky.UnluckCheck) return false;`

**Comment by Ultradragon005** [comment]
image
good progress!

**Comment by Ultradragon005** [comment]
well idk if the website is up-to-date, let's hope so.

**Comment by Tommy-XL** [comment]
All that's left to do is finish:
- [ ] Fool
- [ ] Avanger
- [ ] Aware
- [ ] Bloodlust
- [ ] Diseased
- [ ] Gravestone
- [ ] Guesser
- [ ] Rebound
- [ ] Schizophrenic (DualPersonality in code, need rename)
- [ ] Stubborn
- [ ] Madmate
- [ ] Youtuber

**Comment by Tommy-XL** [comment]
Fix conflicts, pls

**Comment by Ultradragon005** [comment]
I will set it as ready to review so it automaticaly syncs, so don't review yet

**Comment by Ultradragon005** [comment]
now it is ready

File: Patches/PlayerControlPatch.cs [hunk 5]
```diff
@@ -824,41 +776,29 @@ public static bool Prefix(PlayerControl __instance, [HarmonyArgument(0)] PlayerC
         {
             switch (killerSubRole)
             {
-                case CustomRoles.Mare when !Utils.IsActive(SystemTypes.Electrical):
-                    return false;
+                case CustomRoles.Mare:
+                    return Mare.IsLightsOut;
```
Comment (Tommy-XL) [inline Patches/PlayerControlPatch.cs:781]: `CheckMurder` should **not** `return true`, It should always `return false`

File: Patches/PlayerControlPatch.cs [hunk 6]
```diff
@@ -824,41 +776,29 @@ public static bool Prefix(PlayerControl __instance, [HarmonyArgument(0)] PlayerC
...
-                case CustomRoles.Swift when !targetRole.Is(CustomRoles.Pestilence):
-                    target.RpcMurderPlayerV3(target);
-                    if (!Options.DisableShieldAnimations.GetBool())
-                        killer.RpcGuardAndKill(killer);
-                    killer.SetKillCooldown();
-                    target.SetRealKiller(killer);
-                    RPC.PlaySoundRPC(killer.PlayerId, Sounds.KillSound);
-                    return false;
+                case CustomRoles.Swift:
+                    return Swift.OnCheckMurder(killer, target);
```
Comment (Tommy-XL) [inline Patches/PlayerControlPatch.cs:801]: Same

**Pull request merged**